shared/engine/index.js

```diff
@@ -192,7 +192,9 @@ class Engine {
       }
     }
 
-    this.sessionIDToResponse[sessionID] = wrappedResponse
+    if (sessionID) {
```

Comment:
but is there still a response that hasn't be responded to? what's the case when sessionID is null?